File: packages/java/src/com/peruspatial/territorial/ClassifyHazard.java
package com.peruspatial.territorial;

import com.peruspatial.data.TerritorialData;

public final class ClassifyHazard {
  private ClassifyHazard() {}

  public static String classifyHazard(String text) {
    if (text == null) return TerritorialData.DEFAULT_HAZARD;
    String lower = text.toLowerCase();
    for (TerritorialData.Rule rule : TerritorialData.RULES) {
      for (String keyword : rule.keywords()) {
        if (lower.contains(keyword)) return rule.type();
      }
    }
    return TerritorialData.DEFAULT_HAZARD;
  }
}

File: packages/python/peru_spatial_utils/territorial/classify_hazard.py
from ._data import hazards


def classify_hazard(text: str) -> str:
    lowered = text.lower()
    data = hazards()
    for rule in data["rules"]:
        for keyword in rule["keywords"]:
            if keyword in lowered:
                return rule["type"]
    return data["default"]
